### ISSUE TITLE
[pytorch] [feature request] Flatten convenience method

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#7743]
Minor suggestion (trivial to implement in user code, but having it in the library would improve code brevity). The purpose is to flatten specific trailing dimensions by passing negative dimension index.

Can be useful for aggregating across multiple trailing dimensions, before mean/max etc get multiple dimensions support.

Exists in numpy/tensorflow/onnx, but semantics there doesn't allow flattening only specific dimensions.

```python
def flatten(x, dim):
    return x.view(x.size()[:dim] + (-1, ))

flatten(torch.rand(2,3,4,5,6), dim = -2).shape
# (2, 3, 4, 30)
```

**Comment by apaszke** [comment]
Seems nice, could it also flatten multiple dimensions in a contiguous range?

**Comment by vadimkantorov** [comment]
The snippet was just to flatten a few starting or trailing dimensions. If this is implemented into Core, maybe it’s better to rename `dim` to `multi_dims`? to align with recent multiple dimensions support in `sum`

**Comment by vadimkantorov** [comment]
Then maybe a naive support for `multi_dims` can be added to other ops like SoftMax by using `flatten` under the hood, at least if `multi_dims` contains contiguous dimensions (in actual memory layout).

**Comment by chopin57otu** [comment]
The documentation for this function is missing.

**Comment by vadimkantorov** [comment]
It might also be nice to have out-variants for often-copying:
- flatten()
- contiguous()
- reshape()
